tests/run-pass/aligned_utf8_check.rs

```diff
@@ -1,13 +1,6 @@
 fn main() {
     const N: usize = 10;
-
-    let x = vec![0x4141u16; N];
-
-    let mut y: Vec<u8> = unsafe { std::mem::transmute(x) };
-    unsafe { y.set_len(2 * N) };
-
-    println!("{:?}", std::str::from_utf8(&y).unwrap());
-
-    let mut x: Vec<u16> = unsafe { std::mem::transmute(y) };
-    unsafe { x.set_len(N) };
+    let vec = vec![0x4141414141414141u64; N];
+    let content = unsafe { std::slice::from_raw_parts(vec.as_ptr() as *const u8, 8 * N) };
+    println!("{:?}", std::str::from_utf8(content).unwrap());
```

Comment:
Do we also have a more direct test that immediately calls `align_offset`? That seems useful to have.